src/hawkbit-client.c

```diff
@@ -741,19 +742,23 @@ gboolean install_complete_cb(gpointer ptr)
  * feedback and call software_ready_cb() callback on success.
  *
  * @param[in] data Artifact* to process
- * @return NULL is always returned
+ * @return if download thread waited for installation to finish, TRUE if installation and download
+ *         succeeded, FALSE otherwise. If download thread did not wait for installation to finish,
+ *         TRUE if download succeeded, FALSE otherwise.
```

Comment:
Maybe it's also an option to use `GINT_TO_POINTER()` here?

Reply:
Yep!